CrossPlatformDesktopProject/CrossPlatformDesktopProject/Link/Equipables/Bomb.cs

```diff
@@ -0,0 +1,130 @@
+﻿using CrossPlatformDesktopProject.Link;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CrossPlatformDesktopProject.Equipables
+{
+    class Bomb : IEquipable
+    {
+        private Player player;
+        private Vector2 currentPos;
+        private float usedX;
+        private float usedY;
+        private int my_frame_index;
+        private int delay_frame_index;
+        private int delay_frames;
+        private int explode_time;
+        private int current_frame;
+        private int swap_frame;
+        private int swap_frame_index;
+
+        private static List<Rectangle> my_source_frames = new List<Rectangle>
+        {
+            LinkTextureStorage.BOMB,
+            LinkTextureStorage.BOMB_EXPLOSION_1,
+            LinkTextureStorage.BOMB_EXPLOSION_2,
+            LinkTextureStorage.BOMB_EXPLOSION_3
+        };
+
+        public Bomb(Player player)
+        {
+            this.player = player;
+            usedX = player.xPos;
+            usedY = player.yPos;
+            currentPos = new Vector2(usedX, usedY);
+            my_frame_index = 0;
+            delay_frame_index = 0;
+            explode_time = 100;
+            current_frame = 1;
+            swap_frame = 8;
+            swap_frame_index = 0;
+            delay_frames = 5;
+        }
+
+        public void Draw(SpriteBatch spriteBatch)
+        {
+            Texture2D texture = LinkTextureStorage.Instance.getLinkSpriteSheet();
+            Rectangle source = my_source_frames[my_frame_index];
+            Rectangle destination;
+            if (my_frame_index == 0)
+            {
+                destination = new Rectangle((int)currentPos.X, (int)currentPos.Y, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+            } else if (current_frame == 1 || my_frame_index >= 2)
+            {
+                destination = new Rectangle((int)currentPos.X, (int)currentPos.Y, source.Width * 3, source.Height * 3);

```

Comment:
Can this become a foreach over a constant `List<Vector2D>`? Maybe containing the points
`[(0, 0), (48, 0), (24, 38), (-24, -38)]`

Reply:
Maybe even a `Dictionary<int, List<Vector2D>` to avoid the `if` statements.